internal/resourcepolicies/volume_policies.go

```diff
@@ -0,0 +1,180 @@
+package resourcepolicies
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type volumePolicier struct {
+	action        Action
+	volConditions []volumeCondition
+}
+
+type volumeCondition interface {
+	Validate() (bool, error)
+	Match(v *StructuredVolume) bool
+}
+
+type policyConditionsMatcher struct {
+	policies []*volumePolicier
+}
+
+// Capacity consist of the lower and upper boundary
+type Capacity struct {
+	lower resource.Quantity
+	upper resource.Quantity
+}
+
+type StructuredVolume struct {
+	capacity     resource.Quantity
+	storageClass string
+	nfs          *struct{}
+	csi          *csiVolumeSource
+}
+
+type capacityCondition struct {
+	capacity Capacity
+}
+
+func (c *capacityCondition) Match(v *StructuredVolume) bool {
+	return c.capacity.isInRange(v.capacity)
+}
+
+type storageClassCondition struct {
+	storageClass []string
+}
+
+func (s *storageClassCondition) Match(v *StructuredVolume) bool {
+	if len(s.storageClass) == 0 {
+		return true
+	} else if v.storageClass == "" {
```

Comment:
No `else` is needed here, just `if v.storageClass == ""` is OK as you have returned if the last `if` match, please check all other similar codes.

Reply:
modified